src/main/java/de/srendi/advancedperipherals/common/addons/computercraft/apis/EntityAPI.java

```diff
@@ -0,0 +1,101 @@
+package de.srendi.advancedperipherals.common.addons.computercraft.apis;
+
+import dan200.computercraft.api.lua.*;
+import dan200.computercraft.shared.util.NBTUtil;
+import de.srendi.advancedperipherals.common.configuration.APConfig;
+import de.srendi.advancedperipherals.common.util.EntityUtil;
+import de.srendi.advancedperipherals.common.util.LuaConverter;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.player.Player;
+
+import java.util.UUID;
+
+public class EntityAPI {
+
+    public static final EntityAPI ENTITY_API = new EntityAPI();
+
+    @LuaFunction(mainThread = true)
+    public final Object getNBT(IArguments arguments) throws LuaException {
+        if (!APConfig.API_CONFIG.enableGetNBT.get())
+            throw new LuaException("This function is disabled in the config. Activate it or ask an admin if they can activate it.");
+        try {
+            Entity entity = EntityUtil.getEntityFromUUID(UUID.fromString(arguments.getString(0)));
+            if (!APConfig.API_CONFIG.enablePlayerAccess.get() && entity instanceof Player)
+                return MethodResult.of(null, "Using players in EntityAPI is disabled in the config. Activate it or ask an admin if they can activate it.");
+            return NBTUtil.toLua(entity.serializeNBT());
+        } catch (IllegalArgumentException e) {
```

Comment:
I see you want catch the `IllegalArgumentException` error that `getEntityFromUUID` throws.
First of all, not only `getEntityFromUUID` throws IllegalArgumentException, there is a lot internal java methods throws it, e.g. `UUID.fromString`
Secondly, as you can see in the first point, try catch makes program process unclear, you will never know what and where exceptions being throws, even if you defined a specific error `EntityMissingException` you will not know if that exception is formed by the current finding entity missing or if it is because some else entity is missing.

Reply:
Then, why not just make `getEntityFromUUID` returns `null` when entity was not found? You can check the null value easily as well.

Reply:
Sure I'll do that, and add another error check for `UUID.fromString`.

Reply:
You do not need to check the error of `UUID.fromString`, because that is an explicit error, which users should not pass a non-uuid value